up42/initialization.py

```diff
@@ -95,7 +99,11 @@ def initialize_job(job_id: str, project_id: Optional[str] = None) -> "Job":
         project_id: The id of the UP42 project, containing the job
     """
     _check_deprecated_implicit_project_id(project_id)
-    job = Job(auth=main._auth, job_id=job_id, project_id=project_id or str(main._auth.project_id))
+    job = Job(
+        auth=main._auth,
+        job_id=job_id,
+        project_id=project_id or str(main._auth.project_id),
```

Comment:
This is not what I proposed. I meant smith like this:
```python
def _get_project_id(project_id: Optional[str]):
    if not project_id:
        warn(
            "Provide the project ID as the value of the `project_id` argument.",
            DeprecationWarning,
            stacklevel=2,
        )
    result = project_id or str(main._auth.project_id)
    if not result:
        raise ValueError("Project ID can't be null")
    return result
```
and use it everywhere instead of `project_id or str(main._auth.project_id)` combination in all 5 cases